code/WorkInProgress/MechanicMadness.dm

```diff
@@ -3,6 +3,242 @@
+	var/can_be_welded=0
+	var/can_be_anchored=0
+	custom_suicide=1
+	ex_act(severity)
+		switch(severity)
+			if (1.0)
+				src.dispose() // disposing upon being blown up unlike all those decorative rocks on cog2
+				return
+			if (2.0)
+				//tbd, maybe unbolt/open?
+				return
+			if (3.0)
+				//??????
+				return
+		return
+	suicide(var/mob/user as mob) // lel
+		if (!src.user_can_suicide(user))
+			return 0
+		user.visible_message("<span class='alert'><b>[user] stares into the [src], trying to make sense of its function!</b></span>")
+		SPAWN_DBG(3 SECONDS)
+			user.visible_message("<span class='alert'><b>[user]'s brain melts!</b></span>")
+			playsound(get_turf(user), "sound/weapons/phaseroverload.ogg", 100)
+			user.take_brain_damage(69*420)
+		SPAWN_DBG(20 SECONDS)
+			if (user && !isdead(user))
+				user.suiciding = 0
+		return
+
+
+	attackby(obj/item/W as obj, mob/user as mob)
+		if (isscrewingtool(W))
+			if(src.welded)
+				boutput(user,"<span class='alert'>The [src] is welded shut.</span>")
+				return
+			src.does_not_open_in_pocket=src.open
+			src.open=!src.open
+			playsound(src.loc,'sound/items/screwdriver.ogg',50)
+			boutput(user,"<span class='notice'>You [src.open ? "unsecure" : "secure"] the [src]'s cover</span>")
+			src.updateIcon()
+			if(!src.open) src.close_storage_menus()
+			return 1
+		else if (iswrenchingtool(W))
+			if(!src.can_be_anchored)
+				boutput(user,"<span class='alert'>[src] cannot be anchored to the ground.</span>")
+				return
+			if(!src.open || src.welded)
+				boutput(user,"<span class='alert'>You are unable to access the [src]'s bolts as they are on the inside.</span>")
+				return
+			if(!isturf(src.loc) && !src.anchored)
+				boutput(user,"<span class='alert'>You cannot anchor a component housing inside something else.</span>")
+				return
+			src.anchored=!src.anchored
+			playsound(src.loc,'sound/items/Ratchet.ogg',50)
+			boutput(user,"<span class='notice'>You [src.anchored ? "anchor the [src] to" : "unsecure the [src] from"] the ground</span>")
+			if (!src.anchored)
+				src.destroy_outside_connections() //burn those bridges
+			return 1
+		else if (isweldingtool(W))
+			if (!src.can_be_welded)
+				boutput(user,"<span class='alert'>[src]'s cover cannot be welded shut.</span>")
+				return
+			if (src.open)
+				boutput(user,"Why would you want to weld something <i>open?</i>")
+				return
+			if(W:try_weld(user, 1))
+				src.welded=!src.welded
+				boutput(user,"<span class='notice'>You [src.welded ? "" : "un"]weld the [src]'s cover</span>")
+				src.updateIcon()
+				return 1
+		else if (src.open || !istype(W,/obj/item/mechanics))
+			..()
+		return 1
+	proc/updateIcon()
+		if(src.welded)
+			src.icon_state=initial(src.icon_state)+"_w"
+			return
+		if(src.open)
+			src.icon_state=initial(src.icon_state)+"_open"
+		else
+			src.icon_state=initial(src.icon_state)+"_closed"
+		return
+	proc/close_storage_menus() // look for nerds who still have the container UI open and close it for them
```

Comment:
I don't like this system, this ignores teleknesis completely. It's also unperformant and gross. However, you can't easily replace this with `in_range` easily, it would need to be part of a subscriber system.

Reply:
I don't know what exactly you mean by a "subscriber system", I assume it's some kind of event system thing. Can I get an example of it in use somewhere?

Reply:
subscriber system in that: when a user successfully opens the UI, they become "subscribed" to it. They receive updates of it (for example, if another user affects it), and are removed when the UI closes; be that it closed because they pressed the "X", walked away, or the locker was closed itself.
They takes a fair bit of code as Zewaka noted with "you can't easily replace this".
I'm not familiar with any current UI subscriber systems in Goon, so I can't help much more than this description unfortunately.

Reply:
> 
> I'm not familiar with any current UI subscriber systems in Goon, so I can't help much more than this description unfortunately.

tgui lol